### PR TITLE
Derive region for auto-discovered systems

### PR DESCRIPTION
**Pull request by dzahariev**
**Description**
Derive region for auto-discovered systems

Changes proposed in this pull request:
- Fix a bug where optional properties that were not supplied, were processed as empty strings. They should be removed from payload.
- 

**Pull Request status**
- [x] Implementation
- [x] Unit tests
- [x] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [x] Mocks are regenerated, using the automated script

### CODE REVIEW

**Comment by dzahariev** [comment]
reset-vm
/test pre-main-compass-integration-no-dump

File: chart/compass/values.yaml
Comment (dragobt) [inline chart/compass/values.yaml:1]: bump the `e2e_test` PR version as well as you have changes in the e2e tests.

File: components/director/internal/domain/apptemplate/service.go [hunk 1]
```diff
@@ -338,9 +338,61 @@ func (s *service) PrepareApplicationCreateInputJSON(appTemplate *model.Applicati
...
+
+	for _, placeholder := range appTemplate.Placeholders {
+		newValue, err := values.FindPlaceholderValue(placeholder.Name)
+		isOptional := false
+		if placeholder.Optional != nil {
+			isOptional = *placeholder.Optional
+		}
+
+		if err != nil && !isOptional {
+			return "", errors.Wrap(err, "required placeholder not provided")
```
Comment (dragobt) [inline components/director/internal/domain/apptemplate/service.go:350]: add unit test for this case

File: components/director/internal/domain/apptemplate/service.go [hunk 2]
```diff
@@ -338,9 +338,61 @@ func (s *service) PrepareApplicationCreateInputJSON(appTemplate *model.Applicati
...
+		}
```
Comment (dragobt) [inline components/director/internal/domain/apptemplate/service.go:351]: I see that this logic is the same as the one from l323 to l332. Can we somehow do the whole thing in one loop? If not so, perhaps consider taking it out in a separate function.

File: components/director/internal/domain/apptemplate/service.go [hunk 3]
```diff
@@ -338,9 +338,61 @@ func (s *service) PrepareApplicationCreateInputJSON(appTemplate *model.Applicati
...
+
+		if strings.TrimSpace(newValue) == "" && isOptional {
+			appCreateInputJSON, err = removeEmptyKeyFromLabels(appCreateInputJSON, placeholder.Name)
+			if err != nil {
+				return "", errors.Wrap(err, "error while clear optional empty value")
```
Comment (dragobt) [inline components/director/internal/domain/apptemplate/service.go:356]: add unit test for this case as well

File: components/director/internal/domain/apptemplate/service.go (hunk 1 above)
Comment (dzahariev) [reply components/director/internal/domain/apptemplate/service.go:350]: This statement is never reached, as the same check was done several lines above. 
Will look for a way to remove this redundant check.

File: components/director/internal/domain/apptemplate/service.go (hunk 2 above)
Comment (dzahariev) [reply components/director/internal/domain/apptemplate/service.go:351]: Not possible to be done on one pass.
The idea of first pass is to process entries and the second pass require the appCreateInputJSON to be already filled and now we just clear the empty labels from it.

Comment (dzahariev) [reply components/director/internal/domain/apptemplate/service.go:351]: Now I clear the output now on each property process. Slower than before, but penalty can be ignored, as data that is processed is usually les than 15 fields.

File: components/director/internal/domain/apptemplate/service.go [hunk 4]
```diff
@@ -337,10 +337,47 @@ func (s *service) PrepareApplicationCreateInputJSON(appTemplate *model.Applicati
...
+		if _, ok := value.(string); ok {
+			// String value
+			if value == "" && key == keyName {
+				if !rootObject {
+					delete(*input, key)
+				}
+			}
+		} else if mapValue, ok := value.(map[string]interface{}); ok {
+			// Object value - process only labels object
+			if key == "labels" {
```
Comment (dragobt) [inline components/director/internal/domain/apptemplate/service.go:374]: we can take `labels` out as const

**Comment by dzahariev** [comment]
Validated!

**Review by dragobt — APPROVED** [review]

**Pull request merged**